go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -337,12 +339,21 @@ func (oc *Controller) createEgressFirewallRules(priority int, match, action, ext
 		logicalSwitches = append(logicalSwitches, types.OVNJoinSwitch)
 	}
 
+	// a name is needed for logging purposes - the name must be unique, so make it
+	// egressFirewall_<namespace name>_<priority>
+	aclName := buildEgressFwAclName(externalID, priority)
+	log, meter, severity := oc.getLogMeterSeverity(externalID, action)
+
 	egressFirewallACL := &nbdb.ACL{
 		Priority:    priority,
 		Direction:   types.DirectionToLPort,
 		Match:       match,
 		Action:      action,
 		ExternalIDs: map[string]string{"egressFirewall": externalID},
+		Name:        &aclName,
+		Meter:       &meter,
+		Severity:    &severity,
+		Log:         log,
```

Comment:
Do we need to synchronize this on existing firewalls on startup?

Reply:
Yeah, good point. I amended this now.

Not the change that I had to make to go-controller/vendor/github.com/ovn-org/libovsdb/cache/cache.go

I will fix this now in libovsdb (including a unit test) but meanwhile I pushed it here, too. I know that it's not the correct way of fixing this, but without that change, the feature is broken.

~~~
+++ b/go-controller/vendor/github.com/ovn-org/libovsdb/cache/cache.go
@@ -957,7 +957,10 @@ func (t *TableCache) ApplyModifications(tableName string, base model.Model, upda
                        // if NativeToOVS was successful, then simply assign
                        if nv.Type() == reflect.ValueOf(current).Type() {
                                err = info.SetField(k, nv.Interface())
-                               return err
+                               if err != nil {
+                                       return err
+                               }
+                               break
                        }
                        // With a pointer type, an update value could be a set with 2 elements [old, new]
                        if nv.Len() != 2 {
~~~